store/tikv/coprocessor_cache.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tikv
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/dgraph-io/ristretto"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/coprocessor"
+	"github.com/pingcap/tidb/config"
+)
+
+type coprCache struct {
+	cache                   *ristretto.Cache
+	admissionMaxSize        int
+	admissionMinProcessTime time.Duration
+}
+
+type coprCacheValue struct {
+	Data              []byte
+	TimeStamp         uint64
+	RegionID          uint64
+	RegionDataVersion uint64
+}
+
+func (v *coprCacheValue) String() string {
+	return fmt.Sprintf("{ Ts = %d, RegionID = %d, RegionDataVersion = %d, len(Data) = %d }",
+		v.TimeStamp,
+		v.RegionID,
+		v.RegionDataVersion,
+		len(v.Data))
+}
+
+func newCoprCache(config *config.CoprocessorCache) (*coprCache, error) {
+	if config == nil || !config.Enabled {
+		return nil, nil
+	}
+	capacityInBytes := int64(config.CapacityMB * 1024.0 * 1024.0)
+	estimatedEntities := capacityInBytes / int64(config.AdmissionMaxResultMB) * 2
```

Comment:
`bytes/MB`? the unit is different.

Reply:
It is to eliminate the calculation when each cache item is received (receiving byte array).